File: src/main/java/com/fengsheng/handler/skill_qi_huo_ke_ju_tos.java
package com.fengsheng.handler;

import com.fengsheng.HumanPlayer;
import com.fengsheng.protos.Role;
import com.fengsheng.skill.Skill;
import com.fengsheng.skill.SkillId;
import org.apache.log4j.Logger;

public class skill_qi_huo_ke_ju_tos extends AbstractProtoHandler<Role.skill_qi_huo_ke_ju_tos> {
    private static final Logger log = Logger.getLogger(skill_qi_huo_ke_ju_tos.class);

    @Override
    protected void handle0(HumanPlayer r, Role.skill_qi_huo_ke_ju_tos pb) {
        Skill skill = r.findSkill(SkillId.QI_HUO_KE_JU);
        if (skill == null) {
            log.error("你没有这个技能");
            return;
        }
        r.getGame().tryContinueResolveProtocol(r, pb);
    }
}

File: src/main/kotlin/handler/skill_qi_huo_ke_ju_tos.kt
package com.fengsheng.handler

import com.fengsheng.HumanPlayer
import com.fengsheng.protos.Role
import com.fengsheng.skill.SkillId
import org.apache.logging.log4j.kotlin.logger

class skill_qi_huo_ke_ju_tos : AbstractProtoHandler<Role.skill_qi_huo_ke_ju_tos>() {
    override fun handle0(r: HumanPlayer, pb: Role.skill_qi_huo_ke_ju_tos) {
        val skill = r.findSkill(SkillId.QI_HUO_KE_JU)
        if (skill == null) {
            logger.error("你没有这个技能")
            r.sendErrorMessage("你没有这个技能")
            return
        }
        r.game!!.tryContinueResolveProtocol(r, pb)
    }
}